include/solvers/petsc_nonlinear_solver.h

```diff
@@ -153,6 +158,18 @@ class PetscNonlinearSolver : public NonlinearSolver<T>
    */
   void use_default_monitor(bool state) { _default_monitor = state; }
 
+  /**
+   * Petsc solve type
+   */
+  enum SolveType
+  {
+    MF_OPERATOR, ///< Preconditioned Jacobian-free Newton
+    MF,          ///< Jacobian-free Newton
+    STANDARD     ///< Newton with explicit matrix
```

Comment:
Should this be called NEWTON?

Reply:
I see. How about "Matrix Full" Newton, then we can abbreviate it with "MF"... oh wait.

Reply:
"Newton" is a somewhat fuzzy term; I've even heard some insist that the typical way libMesh apps do solves be referred to only as "Newton-Krylov" or "quasi-Newton", to avoid implying that we're doing a solve to machine precision at every (quasi-)Newton step.

I'm happy either way but I think the tie-breaker would be consistency with MOOSE.